File: pythonlibrary/aibrd/workspace/writer.py
import os


def ensure_dir(path: str) -> None:
    os.makedirs(path, exist_ok=True)


def write_file(path: str, content: str) -> None:
    ensure_dir(os.path.dirname(path))
    with open(path, "w", encoding="utf-8") as f:
        f.write(content)


def append_file(path: str, content: str) -> None:
    ensure_dir(os.path.dirname(path))
    with open(path, "a", encoding="utf-8") as f:
        f.write(content)


def init_structure(aibrd_dir: str, modular: bool) -> None:
    ensure_dir(aibrd_dir)
    ensure_dir(os.path.join(aibrd_dir, "shared"))
    ensure_dir(os.path.join(aibrd_dir, "releases"))
    if modular:
        ensure_dir(os.path.join(aibrd_dir, "modules"))


def init_module(aibrd_dir: str, slug: str) -> None:
    ensure_dir(os.path.join(aibrd_dir, "modules", slug))
    ensure_dir(os.path.join(aibrd_dir, "modules", slug, "tests"))


def write_context(aibrd_dir: str, content: str, module_slug: str | None = None) -> None:
    path = (
        os.path.join(aibrd_dir, "modules", module_slug, "CONTEXT.md")
        if module_slug
        else os.path.join(aibrd_dir, "CONTEXT.md")
    )
    write_file(path, content)


def write_tests(aibrd_dir: str, content: str, module_slug: str | None = None) -> None:
    path = (
        os.path.join(aibrd_dir, "modules", module_slug, "tests", "test-cases.md")
        if module_slug
        else os.path.join(aibrd_dir, "tests", "test-cases.md")
    )
    write_file(path, content)


def write_release(aibrd_dir: str, version: str, content: str) -> None:
    write_file(os.path.join(aibrd_dir, "releases", f"{version}.md"), content)

File: src/workspace/writer.ts
import * as fs from 'fs'
import * as path from 'path'

export function ensureDir(dirPath: string): void {
  fs.mkdirSync(dirPath, { recursive: true })
}

export function writeFile(filePath: string, content: string): void {
  ensureDir(path.dirname(filePath))
  fs.writeFileSync(filePath, content, 'utf-8')
}

export function appendToFile(filePath: string, content: string): void {
  ensureDir(path.dirname(filePath))
  fs.appendFileSync(filePath, content, 'utf-8')
}

export function initAibrdStructure(aibrdDir: string, modular: boolean): void {
  ensureDir(aibrdDir)
  ensureDir(path.join(aibrdDir, 'shared'))
  ensureDir(path.join(aibrdDir, 'releases'))
  if (modular) {
    ensureDir(path.join(aibrdDir, 'modules'))
  }
}

export function initModuleStructure(aibrdDir: string, moduleSlug: string): void {
  ensureDir(path.join(aibrdDir, 'modules', moduleSlug))
  ensureDir(path.join(aibrdDir, 'modules', moduleSlug, 'tests'))
}

export function writeContextMd(
  aibrdDir: string,
  content: string,
  moduleSlug?: string
): void {
  const filePath = moduleSlug
    ? path.join(aibrdDir, 'modules', moduleSlug, 'CONTEXT.md')
    : path.join(aibrdDir, 'CONTEXT.md')
  writeFile(filePath, content)
}

export function writeTestCases(
  aibrdDir: string,
  content: string,
  moduleSlug?: string
): void {
  const filePath = moduleSlug
    ? path.join(aibrdDir, 'modules', moduleSlug, 'tests', 'test-cases.md')
    : path.join(aibrdDir, 'tests', 'test-cases.md')
  writeFile(filePath, content)
}

export function writeRelease(
  aibrdDir: string,
  version: string,
  content: string
): void {
  const filePath = path.join(aibrdDir, 'releases', `${version}.md`)
  writeFile(filePath, content)
}

export function updateContextMd(
  contextPath: string,
  section: string,
  newContent: string
): void {
  const existing = fs.existsSync(contextPath)
    ? fs.readFileSync(contextPath, 'utf-8')
    : ''
  const updated = existing + '\n\n' + newContent
  writeFile(contextPath, updated)
}
